helpers/order/order.go

```diff
@@ -17,40 +17,100 @@
 package order
 
 import (
+	"fmt"
+
 	"github.com/softlayer/softlayer-go/datatypes"
 	"github.com/softlayer/softlayer-go/services"
 	"github.com/softlayer/softlayer-go/session"
 )
 
-// CheckBillingOrderStatus returns true if the status of the billing order for
-// the provided product order receipt is in the list of provided statuses.
-// Returns false otherwise, along with the billing order item used to check the statuses,
-// and any error encountered.
-func CheckBillingOrderStatus(sess *session.Session, receipt *datatypes.Container_Product_Order_Receipt, statuses []string) (bool, *datatypes.Billing_Order_Item, error) {
-	service := services.GetBillingOrderItemService(sess)
+// CheckItemProvisionStatus returns true if the status of the provision
+// transaction for ALL of the provided order items is in the list of provided
+// statuses. Returns false otherwise.  The order items provided must be part of
+// the same order.
+//
+// As a side-effect, the original passed order items are updated with the
+// latest billing item details, in an effort to avoid the need for further
+// trips to the API to fetch this info.
+//
+// Returns an error if an order item has no billing item or the billing item
+// has no provision transaction (this probably just means that the billing
+// item has not been created yet). Also returns an error if all order items are
+// not part of the same overall order. In such cases, the function returns
+// immediately, without checking remaining order items.
+func CheckItemProvisionStatus(sess *session.Session, orderItems []datatypes.Billing_Order_Item, statuses []string) (bool, error) {
+	if len(orderItems) == 0 {
+		return false, fmt.Errorf("No order items provided")
+	}
 
-	item, err := service.
-		Id(*receipt.PlacedOrder.Items[0].Id).
-		Mask("mask[id,billingItem[id,provisionTransaction[id,transactionStatus[name]]]]").
-		GetObject()
+	order, err := services.GetBillingOrderItemService(sess).
+		Id(*orderItems[0].Id).
+		Mask("mask[id,items[billingItem[provisionTransaction[id,transactionStatus[name]]]]]").
+		GetOrder()
```

Comment:
The `IN` filter works as you might expect.  It selects which orders are returned (i.e., those having a matching order item), you still get back all items in those orders, so you still have to do some post-query filtering, to only check those items you care about.